File: Go/0026 Remove Duplicates from Sorted Array/26. Remove Duplicates from Sorted Array.go
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/

// Runtime: 8 ms, faster than 90.64% of Go online submissions for Remove Duplicates from Sorted Array.
// Memory Usage: 4.6 MB, less than 100.00% of Go online submissions for Remove Duplicates from Sorted Array.

package main

import "fmt"

func main() {
	nums := []int{1, 1, 2}
	removeDuplicates(nums)
	fmt.Println(nums)
}

func removeDuplicates(nums []int) int {
	if len(nums) == 0 {
		return 1
	}
	unique := 0
	for index := range nums[1:] {
		if nums[index+1] != nums[unique] {
			unique++
			nums[unique] = nums[index+1]
		}
	}
	return unique + 1
}

File: Python/26. Remove Duplicates from Sorted Array.py
# https://leetcode.com/problems/remove-duplicates-from-sorted-array

# Runtime: 84 ms, faster than 76.60% of Python3 online submissions for Remove Duplicates from Sorted Array.
# Memory Usage: 14.7 MB, less than 86.07% of Python3 online submissions for Remove Duplicates from Sorted Array.

class Solution:
    def removeDuplicates(self, nums):        
        unique = 0
        for index in range(1, len(nums)):
            if nums[unique] != nums[index]:
                unique += 1
                nums[unique] = nums[index]
        
        return unique + 1
    
    
nums = [1,1,2]
print(Solution().removeDuplicates(nums))
print(nums)